File: Service/Implement/CompanyInfoDonViDongGoiNongHoService.cs
namespace Service.Implement
{
    public class CompanyInfoDonViDongGoiNongHoService : BaseService<CompanyInfoDonViDongGoiNongHo, ICompanyInfoDonViDongGoiNongHoRepository>
    , ICompanyInfoDonViDongGoiNongHoService
    {
        private readonly ICompanyInfoDonViDongGoiNongHoRepository _CompanyInfoDonViDongGoiNongHoRepository;
        private readonly IThanhVienService _ThanhVienService;
        public CompanyInfoDonViDongGoiNongHoService(ICompanyInfoDonViDongGoiNongHoRepository CompanyInfoDonViDongGoiNongHoRepository
            , IThanhVienService ThanhVienService
        ) : base(CompanyInfoDonViDongGoiNongHoRepository)
        {
            _CompanyInfoDonViDongGoiNongHoRepository = CompanyInfoDonViDongGoiNongHoRepository;
            _ThanhVienService = ThanhVienService;
        }
        public override void Initialization(CompanyInfoDonViDongGoiNongHo model)
        {
            if (model.ThanhVienID > 0)
            {
                ThanhVien thanhVien = _ThanhVienService.GetByID(model.ThanhVienID.Value);
                model.Name = thanhVien.Name;
                model.CCCD = thanhVien.CCCD;
                model.DienThoai = thanhVien.DienThoai;
                model.Email = thanhVien.Email;
            }
        }
    }
}

File: CMS/src/app/shared/CompanyInfoDonViDongGoiNongHo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { CompanyInfoDonViDongGoiNongHo } from 'src/app/shared/CompanyInfoDonViDongGoiNongHo.model';
import { BaseService } from './Base.service';
@Injectable({
    providedIn: 'root'
})
export class CompanyInfoDonViDongGoiNongHoService extends BaseService{

    List: CompanyInfoDonViDongGoiNongHo[] | undefined;
    ListFilter: CompanyInfoDonViDongGoiNongHo[] | undefined;
    FormData!: CompanyInfoDonViDongGoiNongHo;

    DisplayColumns001: string[] = ['ThanhVienID','CCCD','DienThoai','Email', 'Note', 'Save'];

    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "CompanyInfoDonViDongGoiNongHo";
    }
}

